src/main/java/io/gravitee/repository/mongodb/management/MongoNodeMonitoringRepository.java

```diff
@@ -0,0 +1,104 @@
+    private NodeMonitoringMongoRepository internalNodeMonitoringRepository;
+
+    @Override
+    public Maybe<Monitoring> findByNodeIdAndType(String nodeId, String type) {
+        MonitoringMongo monitoring = internalNodeMonitoringRepository.findByNodeIdAndType(nodeId, type);
+        return monitoring != null ? Maybe.just(convert(monitoring)) : Maybe.empty();
+    }
+
+    @Override
+    public Single<Monitoring> create(Monitoring monitoring) {
+        MonitoringMongo monitoringMongo = internalNodeMonitoringRepository.insert(convert(monitoring));
+        return Single.just(convert(monitoringMongo));
+    }
+
+    @Override
+    public Single<Monitoring> update(Monitoring monitoring) {
+        if (monitoring == null || monitoring.getId() == null) {
+            return Single.error(new IllegalStateException("Node monitoring to update must have an id"));
+        }
+
+        MonitoringMongo monitoringMongo = internalNodeMonitoringRepository.findById(monitoring.getId()).orElse(null);
+
+        if (monitoringMongo == null) {
+            return Single.error(new IllegalStateException(
+                    String.format("No node monitoring found with id [%s]", monitoring.getId())));
+        }
+
+        monitoringMongo = convert(monitoring);
+        monitoringMongo = internalNodeMonitoringRepository.save(monitoringMongo);
+        return Single.just(convert(monitoringMongo));
+    }
+
+    @Override
+    public Flowable<Monitoring> findByTypeAndTimeFrame(String type, long from, long to) {
+        return Flowable.fromIterable(
+                internalNodeMonitoringRepository.findByTypeAndTimeFrame(type, from, to)
+                        .stream()
+                        .map(this::convert)
+                        .collect(Collectors.toList()));
+    }
+
+    private Monitoring convert(MonitoringMongo monitoringMongo) {
+        final Monitoring monitoring = new Monitoring();
+        monitoring.setId(monitoringMongo.getId());
+        monitoring.setNodeId(monitoringMongo.getNodeId());
+        monitoring.setType(monitoringMongo.getType());
+        monitoring.setEvaluatedAt(monitoringMongo.getEvaluatedAt());
+        monitoring.setCreatedAt(monitoringMongo.getCreatedAt());
+        monitoring.setUpdatedAt(monitoringMongo.getUpdatedAt());
+        monitoring.setPayload(monitoringMongo.getPayload());
+
+        return monitoring;
+    }
+
+    private MonitoringMongo convert(Monitoring monitoring) {
+        final MonitoringMongo monitoringMongo = new MonitoringMongo();
+        monitoringMongo.setId(monitoring.getId());
+        monitoringMongo.setNodeId(monitoring.getNodeId());
+        monitoringMongo.setType(monitoring.getType());
+        monitoringMongo.setEvaluatedAt(monitoring.getEvaluatedAt());
+        monitoringMongo.setCreatedAt(monitoring.getCreatedAt());
+        monitoringMongo.setUpdatedAt(monitoring.getUpdatedAt());
+        monitoringMongo.setPayload(monitoring.getPayload());
+
+        return monitoringMongo;
+    }
```

Comment:
Maybe you you could use a mapper to avoid these `convert` methods ?

Reply:
done